tests/test_embit_utils.py

```diff
@@ -0,0 +1,319 @@
+
+        # https://github.com/bitcoin/bips/blob/master/bip-0049.mediawiki#test-vectors (first payment address of nested segwit on testnet)
+        (bip32.HDKey.from_string("upub5EFU65HtV5TeiSHmZZm7FUffBGy8UKeqp7vw43jYbvZPpoVsgU93oac7Wk3u6moKegAEWtGNF8DehrnHtv21XXEMYRUocHqguyjknFHYfgY"), "nes", 0, False, "test"):
+            "2Mww8dCYPUpKHofjgcXcBCEGmniw9CoaiD2",
+        # jdlcdl: derived via iancoleman test vector for first change address of nested segwit on testnet
+        (bip32.HDKey.from_string("upub5EFU65HtV5TeiSHmZZm7FUffBGy8UKeqp7vw43jYbvZPpoVsgU93oac7Wk3u6moKegAEWtGNF8DehrnHtv21XXEMYRUocHqguyjknFHYfgY"), "nes", 0, True, "test"):
+            "2MvdUi5o3f2tnEFh9yGvta6FzptTZtkPJC8",
+
+        # https://github.com/bitcoin/bips/blob/master/bip-0086.mediawiki#test-vectors (first payment address of taproot on mainnet)
+        (bip32.HDKey.from_string("xpub6BgBgsespWvERF3LHQu6CnqdvfEvtMcQjYrcRzx53QJjSxarj2afYWcLteoGVky7D3UKDP9QyrLprQ3VCECoY49yfdDEHGCtMMj92pReUsQ"), "tr", 0, False, "main"):
+            "bc1p5cyxnuxmeuwuvkwfem96lqzszd02n6xdcjrs20cac6yqjjwudpxqkedrcr",
+
+        # https://github.com/bitcoin/bips/blob/master/bip-0086.mediawiki#test-vectors (second payment address of taproot on mainnet)
+        (bip32.HDKey.from_string("xpub6BgBgsespWvERF3LHQu6CnqdvfEvtMcQjYrcRzx53QJjSxarj2afYWcLteoGVky7D3UKDP9QyrLprQ3VCECoY49yfdDEHGCtMMj92pReUsQ"), "tr", 1, False, "main"):
+            "bc1p4qhjn9zdvkux4e44uhx8tc55attvtyu358kutcqkudyccelu0was9fqzwh",
+
+        # https://github.com/bitcoin/bips/blob/master/bip-0086.mediawiki#test-vectors (first change address of taproot on mainnet)
+        (bip32.HDKey.from_string("xpub6BgBgsespWvERF3LHQu6CnqdvfEvtMcQjYrcRzx53QJjSxarj2afYWcLteoGVky7D3UKDP9QyrLprQ3VCECoY49yfdDEHGCtMMj92pReUsQ"), "tr", 0, True, "main"):
+            "bc1p3qkhfews2uk44qtvauqyr2ttdsw7svhkl9nkm9s9c3x4ax5h60wqwruhk7",
+
+        # jdlcdl: derived via electrum m/44'/1'/0 (first payment address p2pkh on testnet)
+        (bip32.HDKey.from_string("tpubDC5FSnBiZDMmhiuCmWAYsLwgLYrrT9rAqvTySfuCCrgsWz8wxMXUS9Tb9iVMvcRbvFcAHGkMD5Kx8koh4GquNGNTfohfk7pgjhaPCdXpoba"), "leg", 0, False, "test"):
+            "mkpZhYtJu2r87Js3pDiWJDmPte2NRZ8bJV",
+
+        # jdlcdl: derived via electrum m/44'/1'/0 (first change address p2pkh on testnet)
+        (bip32.HDKey.from_string("tpubDC5FSnBiZDMmhiuCmWAYsLwgLYrrT9rAqvTySfuCCrgsWz8wxMXUS9Tb9iVMvcRbvFcAHGkMD5Kx8koh4GquNGNTfohfk7pgjhaPCdXpoba"), "leg", 0, True, "test"):
+            "mi8nhzZgGZQthq6DQHbru9crMDerUdTKva",
+    }
+    func = embit_utils.get_single_sig_address
+
+    print()
+    for args, expected in vectors_args_expected.items():
+        print("\nasserting...")
+
+        # call without optional params (defaults: script_type="nat", index=0, is_change=False, embit_network="main")
+        if args[1:5] == ("nat", 0, False, "main"):
+            print(f'  {func.__name__}({args[0]}) == "{expected}"')
+            assert str(func(args[0], args[1])) == expected
```

Comment:
This line isn't doing what the comment indicates it should be doing.  should be calling only with args[0].